Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -594,6 +596,33 @@ void GiveThreeInvisibleProblemsSaved_WhenStudentGetProblemsByTagsAndPage_ThenSho
         assertTrue(problems.isEmpty());
     }
 
+    @Test
+    void GiveThreeInvisibleProblemsSaved_WhenGuestGetProblemsByIds_ThenShouldRespondEmptyProblems() throws Exception {
+        Integer[] problemIds = {1, 2, 3};
+        saveProblems(problemIds);
+
+        var problems = getProblemsByGuest(problemIds);
+
+        assertTrue(problems.isEmpty());
+    }
+
+    private List<ProblemView> getProblemsByGuest(Integer... problemIds) throws Exception {
```

Comment:
can you support such method signature with the usage of `getProblems(problemIds, withToken(token))`